From 701708d7972bc5905e40bfd125125ec7d1dbe765 Mon Sep 17 00:00:00 2001
From: Elizabeth Figura <zfigura@codeweavers.com>
Date: Tue, 21 Oct 2025 13:46:38 -0500
Subject: [PATCH 45/48] mountmgr.sys: Report FILE_SUPPORTS_REPARSE_POINTS for
 volumes we report as NTFS.

(cherry picked from commit a62285c2dafb16520d83700c89be6481d1763df7)

CW-Bug-Id: #26492
---
 dlls/mountmgr.sys/device.c | 2 +-
 dlls/ntdll/tests/file.c    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/mountmgr.sys/device.c b/dlls/mountmgr.sys/device.c
index d0308a3c993..fc9102f10ea 100644
--- a/dlls/mountmgr.sys/device.c
+++ b/dlls/mountmgr.sys/device.c
@@ -1734,7 +1734,7 @@ static NTSTATUS WINAPI harddisk_query_volume( DEVICE_OBJECT *device, IRP *irp )
             break;
         default:
             fsname = L"NTFS";
-            info->FileSystemAttributes = FILE_CASE_PRESERVED_NAMES | FILE_PERSISTENT_ACLS;
+            info->FileSystemAttributes = FILE_CASE_PRESERVED_NAMES | FILE_PERSISTENT_ACLS | FILE_SUPPORTS_REPARSE_POINTS;
             info->MaximumComponentNameLength = 255;
             break;
         }
diff --git a/dlls/ntdll/tests/file.c b/dlls/ntdll/tests/file.c
index 7c640d225ae..dcd93213da3 100644
--- a/dlls/ntdll/tests/file.c
+++ b/dlls/ntdll/tests/file.c
@@ -6224,7 +6224,7 @@ static void test_reparse_points(void)
 
     status = GetVolumeInformationW( L"C:\\", NULL, 0, NULL, NULL, &flags, NULL, 0 );
     ok( status == TRUE, "got error %lu\n", GetLastError() );
-    todo_wine ok( flags & FILE_SUPPORTS_REPARSE_POINTS, "C: drive does not support reparse points\n" );
+    ok( flags & FILE_SUPPORTS_REPARSE_POINTS, "C: drive does not support reparse points\n" );
 
     swprintf( path, ARRAY_SIZE(path), L"\\??\\%s", temp_path );
     RtlInitUnicodeString( &nameW, path );
-- 
2.53.0

